itou/metabase/management/commands/sql/005_suivi_saisies_dans_asp.sql

```diff
@@ -43,11 +42,17 @@ FROM
         left join "fluxIAE_AnnexeFinanciere_v2" af
             on emi.emi_afi_id = af.af_id_annexe_financiere
             /* Ne prendre en compte que les structures qui ont une annexe financière 
-            valide  pour l'année en cours */
+            valide  pour l'année en cours et prendre en compte les structures qui sont en retard de saisie dans l'asp */
             and af_etat_annexe_financiere_code in ('VALIDE', 'SAISI')
-            and date_part('year', to_date(af.af_date_debut_effet, 'dd/mm/yyyy')) = date_part('year', current_date)
-            /* Ne prendre que les déclarations mensuelles de l'année en cours */
-            and  emi.emi_sme_annee = date_part('year', current_date)
+            and date_part('year', af.af_date_debut_effet_v2) >= (date_part('year', current_date) - 1)
+            and af.af_date_fin_effet_v2 
+                    >= 
+                    make_date(
```

Comment:
J'aimerais bien un commentaire stp qui explique ce que ce `make_date` est supposé retourner.

Reply:
On peut faire plus simple et plus clair avec`and af.af_date_fin_effet_v2 >= CURRENT_DATE - INTERVAL '3 months'`
Exemple: Pour une structure s qui avait une annexe financière valide pour 2021 (du 01/01 au 31/12) et une autre pour 2022.
Sans le filtre sur `af.af_date_fin_effet_v2 ` on va se retrouver dans le tableau de bord Metabase avec les 2 numéros d'af tout au long de l'année 2022.
Avec le filtre sur `af.af_date_fin_effet_v2 ` on affichera les données de l'af 2021 et de l'af 2022 jusqu'à Mars 2022 (on donne 3 mois pour visualiser les stats de l'af de l'année dernière) et ensuite on aura que la data de l'af 2022 car l'objectif  est de visualiser pour chaque structure les données de l'AF (conventionnement) de l'année en cours et non l'historique